lib/SILOptimizer/Mandatory/AddressLowering.cpp

```diff
@@ -2979,6 +3011,31 @@ class DefRewriter : SILInstructionVisitor<DefRewriter> {
     assert(!storage.storageAddress || storage.storageAddress == addr);
     storage.storageAddress = addr;
   }
+
+  void visitUnconditionalCheckedCastInst(
+      UnconditionalCheckedCastInst *uncondCheckedCast) {
+    SILValue srcVal = uncondCheckedCast->getOperand();
+    assert(srcVal->getType().isLoadable(*pass.function));
+    // Create a stack temporary to store the srcVal
+    SILBuilderWithScope allocBuilder(uncondCheckedCast);
```

Comment:
I don't think you actually need to create any builders in this function. Everything is emitted at the same scope/loc.